pkg/kubelet/container/sync_result.go

```diff
@@ -19,14 +19,62 @@ package container
 import (
 	"errors"
 	"fmt"
+	"time"
 
 	utilerrors "k8s.io/apimachinery/pkg/util/errors"
 )
 
 // TODO(random-liu): We need to better organize runtime errors for introspection.
 
-// ErrCrashLoopBackOff returned when a container Terminated and Kubelet is backing off the restart.
-var ErrCrashLoopBackOff = errors.New("CrashLoopBackOff")
+// BackoffError should be used whenever an error needs to specify a particular backoff duration
+// to the Kubelet.
+type BackoffError struct {
+	err         error
+	backoffTime time.Duration
+}
+
+func NewBackoffError(err error, backoffTime time.Duration) *BackoffError {
+	return &BackoffError{
+		err:         err,
+		backoffTime: backoffTime,
+	}
+}
+
+func (e *BackoffError) Error() string {
+	return e.err.Error()
+}
```

Comment:
optional: you could just embed the error in the backoff struct, and omit this method. i.e.

```go
type BackoffError struct {
  error
  backoffTime time.Duration
}
```

Reply:
Done!